nixos/modules/services/web-servers/telnet.nix

```diff
@@ -0,0 +1,57 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.telnet;
+
+in
+{
+
+  options.services.telnet = {
+
+    enable = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Enable telnetd.
```

Comment:
I am not sure if we should put a warning here since due missing encryption telnet should be only used in trusted networks. Also your mininet example might be about man-in-the-middle attacks on telnet.